third_party/3/dateutil/parser.pyi

```diff
@@ -0,0 +1,75 @@
+from typing import List, Tuple, Optional, Callable
+from datetime import datetime
+
+__all__ = ...  # type: List[str]
+
+
+class parserinfo(object):
+    JUMP = ...  # type: List[str]
+    WEEKDAYS = ...  # type: List[Tuple[str, str]]
+    MONTHS = ...  # type: List[Tuple[str, str]]
+    HMS = ...  # type: List[Tuple[str, str, str]]
+    AMPM = ...  # type: List[Tuple[str, str, str]]
+    UTCZONE = ...  # type: List[str]
+    PERTAIN = ...  # type: List[str]
+    TZOFFSET = ...  # type: Dict[str, int]
+
+    def __init__(self, dayfirst: bool=..., yearfirst: bool=...) -> None:
+        ...
+
+    def jump(self, name: str) -> bool:
+        ...
+
+    def weekday(self, name: str) -> str:
+        ...
+
+    def month(self, name: str) -> str:
+        ...
+
+    def hms(self, name: str) -> str:
+        ...
+
+    def ampm(self, name: str) -> str:
+        ...
+
+    def pertain(self, name: str) -> bool:
+        ...
+
+    def utczone(self, name: str) -> bool:
+        ...
+
+    def tzoffset(self, name: str) -> int:
+        ...
+
+    def convertyear(self, year: int) -> int:
+        ...
+
+    def validate(self, year: datetime) -> bool:
+        ...
+
+
+class parser(object):
+    def __init__(self, info: parserinfo=...) -> None:
+        ...
+
+    def parse(
+        self,
+        timestr: str,
+        default: Optional[datetime],
+        ignoretz: bool=...,
+        tzinfos =...,
+            ) -> datetime:
+        ...
+
+DEFAULTPARSER = ...  # type: parser
+
+
+def parse(timestr, parserinfo: parserinfo=..., **kwargs) -> datetime:
+    ...
+
+
+class _tzparser(object):
```

Comment:
`_tzparser` has behavior, but as far as I know it's exclusively used in [`tzstr`](https://github.com/dateutil/dateutil/blob/master/dateutil/tz/tz.py#L702) for parsing a string to a `_tzparser._result` object.